CMakeLists.txt

```diff
@@ -28,6 +28,10 @@ enable_language(CXX)
 #####################################################################
 
 include(CheckCXXCompilerFlag)
+check_cxx_compiler_flag("-std=c++11" COMPILER_SUPPORTS_CXX11)
+if(COMPILER_SUPPORTS_CXX11)
+  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -std=gnu++0x")
```

Comment:
To be honest, I also do not understand why this options have been added. I tried 

set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11") instead and it works.